black_it/loss_functions/base.py

```diff
@@ -62,6 +62,45 @@ def compute_loss(
         """
         num_coords = real_data.shape[1]
 
+        weights = self._check_coordinate_weights(num_coords)
+        filters = self._check_coordinate_filters(num_coords)
+
+        filtered_data = self._filter_data(num_coords, filters, sim_data_ensemble)
+
+        loss = 0
+        for i in range(num_coords):
+            loss += self.compute_loss_1d(filtered_data[i], real_data[:, i]) * weights[i]
+
+        return loss
+
+    @staticmethod
+    def _filter_data(
+        num_coords: int,
```

Comment:
If suggestion below is accepted, `num_coords` is not needed here and above at line 68:

```suggestion
```